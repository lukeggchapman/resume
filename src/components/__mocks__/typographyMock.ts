import idObj from 'identity-obj-proxy'

jest.doMock('../typography', () => ({
  __esModule: true,
  default: idObj,
  FAIcon: 'FAIcon',
}))
