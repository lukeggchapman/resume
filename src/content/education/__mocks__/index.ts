const mockEducationData = [
  {
    frontmatter: {
      school: 'schoolTest1',
      endDate: '2004-05-18',
      degree: 'degreeTest1',
      startDate: '2001-01-11',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
]

export default mockEducationData
