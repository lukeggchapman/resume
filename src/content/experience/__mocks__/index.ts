const mockExperienceData = [
  {
    frontmatter: {
      company: 'companyTest1',
      endDate: 'endDateTest1',
      role: 'roleTest1',
      startDate: 'startDateTest1',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
  {
    frontmatter: {
      company: 'companyTest2',
      endDate: 'endDateTest2',
      role: 'roleTest2',
      startDate: 'startDateTest2',
    },
    logo: 'logoPublcUrlTest2',
    htmlAst: 'htmlAstTest2' as any,
  },
]

export default mockExperienceData
