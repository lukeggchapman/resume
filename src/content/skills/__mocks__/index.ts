export default {
  beginner: ['skillTest1', 'skillTest2'],
  intermediate: ['skillTest3', 'skillTest4'],
  advanced: ['skillTest5', 'skillTest6'],
}
