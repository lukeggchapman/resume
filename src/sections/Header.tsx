import React from 'react'
import { StyleSheet, View, Text, Link as PDFLink } from '@react-pdf/renderer'
import typography, { FAIcon } from 'components/typography'

const stylesObj = {
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 10,
  },
  info: {
    alignItems: 'right',
    fontFamily: 'PT Sans',
  },
}
const styles = StyleSheet.create(stylesObj)

export default () => (
  <View style={styles.header}>
    <Text style={typography.h1}>Luke Chapman</Text>
    <View style={styles.info}>
      <Text>
        <FAIcon icon="envelope" fas />
        {` `}
        <PDFLink src="mailto:lukeggchapman@gmail.com">
          LukeGGChapman@gmail.com
        </PDFLink>
      </Text>
      <Text>
        <FAIcon icon="phone" fas />
        {` `}
        <PDFLink src="tel:+61401516361">0401 516 361</PDFLink>
      </Text>
    </View>
  </View>
)
