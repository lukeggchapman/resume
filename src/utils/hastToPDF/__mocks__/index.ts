export default jest.fn((htmlAst) => htmlAst)
