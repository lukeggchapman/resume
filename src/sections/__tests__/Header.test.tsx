import React from 'react'
import { shallow } from 'enzyme'
import 'components/__mocks__/typographyMock'

import Header from '../Header'

describe('Header Section', () => {
  const render = () => shallow(<Header />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
