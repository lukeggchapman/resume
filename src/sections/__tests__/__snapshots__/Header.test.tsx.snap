// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Section renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
    }
  }
>
  <Text
    style="h1"
  >
    Luke Chapman
  </Text>
  <View
    style={
      Object {
        "alignItems": "right",
        "fontFamily": "PT Sans",
      }
    }
  >
    <Text>
      <FAIcon
        fas={true}
        icon="envelope"
      />
       
      <Link
        src="mailto:lukeggchapman@gmail.com"
      >
        LukeGGChapman@gmail.com
      </Link>
    </Text>
    <Text>
      <FAIcon
        fas={true}
        icon="phone"
      />
       
      <Link
        src="tel:+61401516361"
      >
        0401 516 361
      </Link>
    </Text>
  </View>
</View>
`;
